drivers/accel/adxl38x/adxl38x.c

```diff
@@ -0,0 +1,992 @@
+			j += 2;
+		} else {
+			array_rearranged_data[i] = NULL;
+			array_rearranged_data[i+1] = NULL;
+		}
+		tzyx >>= 1;
+	}
+	*raw_x = adxl38x_accel_array_conv(array_rearranged_data);
+	*raw_y = adxl38x_accel_array_conv(array_rearranged_data + 2);
+	*raw_z = adxl38x_accel_array_conv(array_rearranged_data + 4);
+	*raw_temp = adxl38x_accel_array_conv(array_rearranged_data + 6);
+
+	return ret;
+}
+
+
+/***************************************************************************//**
+ * @brief Reads the raw output data of each axis and converts it to g.
+ *
+ * @param dev  - The device structure.
+ * @param channels - Channels to enable.
+ * @param x    - X-axis's output data.
+ * @param y    - Y-axis's output data.
+ * @param z    - Z-axis's output data.
+ *
+ * @return ret - Result of the reading procedure.
+*******************************************************************************/
+int adxl38x_get_xyz_gees(struct adxl38x_dev *dev,
+			 enum adxl38x_ch_select channels,
+			 struct adxl38x_fractional_val *x, struct adxl38x_fractional_val *y,
+			 struct adxl38x_fractional_val *z)
+{
+	uint16_t raw_accel_x;
+	uint16_t raw_accel_y;
+	uint16_t raw_accel_z;
+	uint16_t raw_accel_t;
+	uint16_t dum;
+	int ret;
+
+	int64_t dummy;
+
+	ret = adxl38x_get_raw_data(dev, channels, &raw_accel_x, &raw_accel_y,
+				   &raw_accel_z, &raw_accel_t);
+
+	if (ret)
+		return ret;
+
+	x->integer = no_os_div_s64_rem((int64_t)adxl38x_accel_conv(dev, raw_accel_x),
+				       ADXL38x_ACC_SCALE_FACTOR_GEE_DIV, &(x->fractional));
+	y->integer = no_os_div_s64_rem((int64_t)adxl38x_accel_conv(dev, raw_accel_y),
+				       ADXL38x_ACC_SCALE_FACTOR_GEE_DIV, &(y->fractional));
+	z->integer = no_os_div_s64_rem((int64_t)adxl38x_accel_conv(dev, raw_accel_z),
+				       ADXL38x_ACC_SCALE_FACTOR_GEE_DIV, &(z->fractional));
+
+	return ret;
+}
+
+
+/***************************************************************************//**
+ * @brief Converts array of raw acceleration to uint16 data raw acceleration
+ *
+ * @param dev       - The device structure.
+ * @param raw_array - Raw acceleration array.
+ *
+ * @return ret      - Converted data.
+*******************************************************************************/
+static uint32_t adxl38x_accel_array_conv(uint8_t *raw_array)
+{
+	return raw_array[1] | ((uint16_t)raw_array[0] << 8);
+}
+
+/***************************************************************************//**
+ * @brief Converts raw acceleration value to m/s^2 value.
+ *
+ * @param dev       - The device structure.
+ * @param raw_accel - Raw acceleration value.
+ *
+ * @return ret      - Converted data.
+*******************************************************************************/
+static int64_t adxl38x_accel_conv(struct adxl38x_dev *dev,
+				  uint16_t raw_accel)
+{
+	int32_t accel_data;
+
+	// Raw acceleration is in two's complement
+	// Convert from two's complement to int
+	if ((raw_accel & NO_OS_BIT(15)) == NO_OS_BIT(15))
```

Comment:
raw_accel & NO_OS_BIT(15) is enough, however, you can use no_os_sign_extend16 for this purpose